src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java

```diff
@@ -98,6 +98,31 @@ public String getVisibilityFilter() {
         return this.visibilityFilter;
     }
 
+    // projectName may be a
+    private String getAid(@Nonnull Run<?,?> run, @Nonnull FilePath workspace, @Nonnull String projectName) throws java.io.IOException {
```

Comment:
Can we use an optional, `@DataBoundSetter` property to allow the user to specify a CodeSonar proj_files directory?  That way, we wouldn't need to assume its location.

It might be easier to provide an optional property to specify the AID.  That would put the burden on the user to find the AID themselves (perhaps by looking in the prj_files directory themselves), but it seems like the plugin code would be less fragile if we did it this way.

We could also provide both options - AID or prj_files directory path, but if both were specified at once, you'd need to raise an exception.

Basically, I think adding an AID option is low risk, but it is more difficult for the end-user.  Adding a prj_files directory option would be easier for the end-user, but it would be a higher risk feature.  I think if you go through the trouble of writing a prj_files option, then an AID option would be very easy and it might be a good fallback in case there are problems with the prj_files option.

Reply:
I can do that, but I fear it might cause some confusion. In the manual, under **Build/Analysis Commands**, there are some pointers that I should remember to include, e.g., the path/to parameter for the analyze command.

Most likely, the chosen default in the function is fine in most cases, I think.